File: SortCheck.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class SortCheck {

	/**
	 * @param args
	 */
	public static void main(String[] args) {

		int counter = 0;
		boolean sorted = true;
		File file = new File(args[0]);
		System.out.println("Elements found in file: ");
		try {
			Scanner in = new Scanner(file);

			int i = in.nextInt();
			System.out.println("S_" + counter + ": " + i);
			counter++;
			while (in.hasNextInt()) {
				int j = in.nextInt();
				System.out.println("S_" + counter + ": " + j);
				counter++;
				if (j < i) {
					System.out
							.println(" Sorting Error: value " + j + " < " + i);
					sorted = false;
				} else
					i = j;
			} // whileloop
			
			in.close();
			
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} // try-catch
		if (sorted)
			System.out.println("File " + args[0] + " is sorted.");
		else
			System.out.println("File " + args[0]
					+ " has not been properly sorted.");

	} // main
} // SortCheck

File: SortCheck.cpp
// SortCheck.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include <iostream>
#include <fstream>
#include <sstream>
using namespace std;

int main(int argc, char* argv[])
{

	int counter = 0;
	bool sorted = true;
	cout << "Filename: ";
	string file;
	getline(cin,file);
	ifstream in(file);

	int i = 0, j = 0;
	in >> i;
	cout << "S_" << counter << ": " << i << endl;
	counter++;
	while (in >> j) {
		cout << "S_" << counter << ": " << j << endl;
		counter++;
		if (j < i) {
			cout << " Sorting Error: value " << j << " < " << i << endl;
			sorted = false;
		}
		else i = j;
	}
	if (sorted) cout << "File " << file << " is sorted." << endl;
	else cout << "File " << file << " has not been properly sorted." << endl;
	return 0;
}

